### PR TITLE
react-flag-icon-css

### PR DESCRIPTION
**Pull request by jonfreedman**
Sharing my definitions for react-flag-icon-css as the project team do not currently maintain their own types (https://github.com/matteocng/react-flag-icon-css/issues/56).  Constructed using docs.

### CODE REVIEW

File: types/react-flag-icon-css/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// Project: https://github.com/matteocng/react-flag-icon-css#readme
+// Definitions by: Jon Freedman <https://github.com/jonfreedman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import { PureComponent } from 'react';
+
+export type Size = 'lg' | '2x' | '3x' | '4x' | '5x';
+
+export interface Options {
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: Can you name this as FlagIconOptions instead to make it clear ?

File: types/react-flag-icon-css/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Type definitions for react-flag-icon-css 1.0
+// Project: https://github.com/matteocng/react-flag-icon-css#readme
+// Definitions by: Jon Freedman <https://github.com/jonfreedman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import { PureComponent } from 'react';
+
+export type Size = 'lg' | '2x' | '3x' | '4x' | '5x';
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: name it as FlagIconSize

File: types/react-flag-icon-css/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export interface Options {
+    useCssModules: boolean;
+}
+
+export interface FlagIconProps {
+    code: string;
+    size?: Size;
+    flip?: 'horizontal' | 'vertical';
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: Type as type FlagIconFlip = 'horizontal' | 'veritcal'

File: types/react-flag-icon-css/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    rotate?: 30 | 60 | 90 | 180 | 270;
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: consder naming type as FlagIconRotate = ... to follow the documentation https://github.com/matteocng/react-flag-icon-css#flags-flagicon-props

File: types/react-flag-icon-css/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    squared?: boolean;
+    className?: string;
+    styleName?: string;
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: What about Component and children on props ?

File: types/react-flag-icon-css/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+// Definitions by: Jon Freedman <https://github.com/jonfreedman>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+import { PureComponent } from 'react';
+
+export type Size = 'lg' | '2x' | '3x' | '4x' | '5x';
+
+export interface Options {
+    useCssModules: boolean;
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/index.d.ts]: Shoul,d this contain customCodes and themeStyles ? https://github.com/matteocng/react-flag-icon-css#factory-flagiconfactory-options

File: types/react-flag-icon-css/tslint.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/react-flag-icon-css/tslint.json]: Please remove anything apart from extends: dtslint/dt.json when writing new defintion

File: types/react-plotly.js/tslint.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/react-plotly.js/tslint.json]: Same here ... tslint.json shouldnt contain anyuthing apart from extends rule

File: types/react-plotly.js/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    data: [{ x: Plotly.Datum, y: Plotly.Datum }];
+    group: 'lower' | 'upper';
+}
+
+export interface Figure {
+    data: Plotly.Data[];
+    layout: Partial<Plotly.Layout>;
+}
+
+export interface PlotParams {
```
Comment (sheetalkamat) [inline types/react-plotly.js/index.d.ts]: Please name this as Props  instead.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/react-flag-icon-css/index.d.ts (hunk 5 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: It looks like these props have been added recently, I will include them

File: types/react-flag-icon-css/tslint.json (hunk 7 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/tslint.json]: I don't agree with this given _Common Mistakes_

> Formatting: Either use all tabs, or always use 4 spaces.

File: types/react-flag-icon-css/index.d.ts (hunk 2 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-flag-icon-css/index.d.ts (hunk 1 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-flag-icon-css/index.d.ts (hunk 6 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-flag-icon-css/index.d.ts (hunk 3 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-flag-icon-css/index.d.ts (hunk 4 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-flag-icon-css/index.d.ts (hunk 5 above)
Comment (jonfreedman) [reply types/react-flag-icon-css/index.d.ts]: done

File: types/react-plotly.js/index.d.ts (hunk 9 above)
Comment (jonfreedman) [reply types/react-plotly.js/index.d.ts]: removed from pr

File: types/react-plotly.js/tslint.json (hunk 8 above)
Comment (jonfreedman) [reply types/react-plotly.js/tslint.json]: removed from pr

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
But the guideline is to not edit generated tslint so please keep it as was being generated when writing for declarations for new package.

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**